didman/api/v1/api.go

```diff
@@ -162,72 +169,55 @@ func (w *Wrapper) GetCompoundServices(ctx echo.Context, didStr string) error {
 	}
 
-	if len(strings.TrimSpace(request.Type)) == 0 {
-		return core.InvalidInputError("invalid value for type")
+	props := request.Body
+	if len(strings.TrimSpace(props.Type)) == 0 {
+		return nil, core.InvalidInputError("invalid value for type")
 	}
 
 	// The api accepts a map[string]interface{} which must be converted to a map[string]ssi.URI.
-	references := make(map[string]ssi.URI, len(request.ServiceEndpoint))
-	for key, value := range request.ServiceEndpoint {
+	references := make(map[string]ssi.URI, len(props.ServiceEndpoint))
+	for key, value := range props.ServiceEndpoint {
 		uri, err := interfaceToURI(value)
 		if err != nil {
-			return core.InvalidInputError("invalid reference for service '%s': %v", key, err)
+			return nil, core.InvalidInputError("invalid reference for service '%s': %v", key, err)
 		}
 		references[key] = *uri
 	}
 
 	// Call Didman
-	service, err := w.Didman.AddCompoundService(ctx.Request().Context(), *id, request.Type, references)
+	service, err := w.Didman.AddCompoundService(ctx, *id, props.Type, references)
 	if err != nil {
-		return err
-	}
-
-	serviceEndpoints, ok := service.ServiceEndpoint.(map[string]interface{})
-	if !ok {
-		return fmt.Errorf("unable to convert service endpoints")
-	}
-
-	cs := CompoundService{
-		Id:              service.ID.String(),
-		ServiceEndpoint: serviceEndpoints,
-		Type:            service.Type,
+		return nil, err
 	}
-	return ctx.JSON(200, cs)
+	return AddCompoundService200JSONResponse(*service), nil
 }
 
 // GetCompoundServiceEndpoint handles calls to read a specific endpoint of a compound service.
-func (w *Wrapper) GetCompoundServiceEndpoint(ctx echo.Context, didStr string, compoundServiceType string, endpointType string, params GetCompoundServiceEndpointParams) error {
-	acceptHeader := ctx.Request().Header.Get("Accept")
-	id, err := did.ParseDID(didStr)
+func (w *Wrapper) GetCompoundServiceEndpoint(_ context.Context, request GetCompoundServiceEndpointRequestObject) (GetCompoundServiceEndpointResponseObject, error) {
+	id, err := did.ParseDID(request.Did)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	resolve := true
-	if params.Resolve != nil {
-		resolve = *params.Resolve
+	if request.Params.Resolve != nil {
+		resolve = *request.Params.Resolve
 	}
-	endpoint, err := w.Didman.GetCompoundServiceEndpoint(*id, compoundServiceType, endpointType, resolve)
+	endpoint, err := w.Didman.GetCompoundServiceEndpoint(*id, request.CompoundServiceType, request.EndpointType, resolve)
 	if err != nil {
-		return err
-	}
-	if strings.Contains(acceptHeader, "text/plain") {
-		return ctx.String(http.StatusOK, endpoint)
+		return nil, err
 	}
 
-	// default json
-	return ctx.JSON(http.StatusOK, EndpointResponse{Endpoint: endpoint})
+	if request.Params.Accept == nil || *request.Params.Accept != "text/plain" {
+		return GetCompoundServiceEndpoint200JSONResponse{Endpoint: endpoint}, nil
+	} else {
+		return GetCompoundServiceEndpoint200TextResponse(endpoint), nil
+	}
```

Comment:
explicitly test for `text/plain` or default to json outside the if statement reduces cognitive complexity